### PR TITLE
Fix README example

### PR DESCRIPTION
**Pull request by rauchg**
- Fixed a warning
- Fixed a bunch of errors
- Applied `rustfmt` (nightly)

Verified that it works with my CosmosDB account

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,8 +46,7 @@ use std::error::Error;
 use futures::future::*;
 use tokio_core::reactor::Core;
 
-use azure_sdk_for_rust::cosmos::{AuthorizationToken, TokenType, Client, PartitionKey};
-
+use azure_sdk_for_rust::cosmos::{AuthorizationToken, Client, TokenType};
```
Comment (rauchg) [inline README.md]: Remove unused `PartitionKey` (warning)

File: README.md [hunk 2]
```diff
@@ -61,11 +60,9 @@ struct MySampleStruct<'a> {
     a_timestamp: i64,
 }
 
-
```
Comment (rauchg) [inline README.md]: `rustfmt`

File: README.md [hunk 3]
```diff
@@ -61,11 +60,9 @@ struct MySampleStruct<'a> {
...
 const DATABASE: &'static str = "azuresdktestdb";
 const COLLECTION: &'static str = "azuresdktc";
 
-
```
Comment (rauchg) [inline README.md]: `rustfmt`

File: README.md [hunk 4]
```diff
@@ -84,28 +81,29 @@ fn code() -> Result<(), Box<Error>> {
     // Let's get Cosmos account and master key from env variables.
     // This helps automated testing.
     let master_key =
-        std::env::var("COSMOS_MASTER_KEY").expect("Set env variable COSMOS_MASTER_KEY first!");
-    let account = std::env::var("COSMOS_ACCOUNT").expect("Set env variable COSMOS_ACCOUNT first!");
+        std::env::var("COSMOSDB_MASTER_KEY").expect("Set env variable COSMOS_MASTER_KEY first!");
+    let account =
+        std::env::var("COSMOSDB_ACCOUNT").expect("Set env variable COSMOS_ACCOUNT first!");
```
Comment (rauchg) [inline README.md]: `rustfmt`

File: README.md [hunk 5]
```diff
@@ -84,28 +81,29 @@ fn code() -> Result<(), Box<Error>> {
...
 
     // First, we create an authorization token. There are two types of tokens, master and resource
     // constrained. Please check the Azure documentation for details. You can change tokens
     // at will and it's a good practice to raise your privileges only when needed.
-    let authorization_token = AuthorizationToken::new(account, TokenType::Master, master_key)?;
+    let authorization_token = AuthorizationToken::new(account, TokenType::Master, &master_key)?;
```
Comment (rauchg) [inline README.md]: According to the [docs](https://docs.rs/azure_sdk_for_rust/0.9.0/azure_sdk_for_rust/cosmos/struct.AuthorizationToken.html), `master_key` is borrowed

File: README.md [hunk 6]
```diff
@@ -84,28 +81,29 @@ fn code() -> Result<(), Box<Error>> {
...
     // list_databases will give us the databases available in our account. If there is
     // an error (for example, the given key is not valid) you will receive a
     // specific AzureError. In this example we will look for a specific database
     // so we chain a filter operation.
-    let future = client.list_databases().and_then(|databases| {
-        ok(databases.into_iter().find(|db| db.id == DATABASE))
-    });
+    let future = client
+        .list_databases()
+        .and_then(|databases| ok(databases.into_iter().find(|db| db.id == DATABASE)));
```
Comment (rauchg) [inline README.md]: `rustfmt` (which did a great job here 😋)

File: README.md [hunk 7]
```diff
@@ -119,7 +117,7 @@ fn code() -> Result<(), Box<Error>> {
     // we will create it. The collection creation is more complex and
     // has many options (such as indexing and so on).
     let collection = {
-        let collections = core.run(client.list_collections(&database))?;
+        let collections = core.run(client.list_collections(&DATABASE))?;
```
Comment (rauchg) [inline README.md]: `list_collections` takes the database name, not the instance. However, I wonder why this changed?

File: README.md [hunk 8]
```diff
@@ -148,7 +145,7 @@ fn code() -> Result<(), Box<Error>> {
             // Performance levels have price impact. Also, higher
             // performance levels force you to specify an indexing
             // strategy. Consult the documentation for more details.
-            core.run(client.create_collection(&database, 400, &coll))?
+            core.run(client.create_collection(&DATABASE, 400, &coll))?
```
Comment (rauchg) [inline README.md]: Same as above

File: README.md [hunk 9]
```diff
@@ -169,12 +166,14 @@ fn code() -> Result<(), Box<Error>> {
     // The method create_document will return, upon success,
     // the document attributes.
     let document_attributes = core.run(
-        client.create_document(&database, &collection, &doc).unwrap()
+        client
+            .create_document(&DATABASE, &COLLECTION, &doc)
+            .execute(),
```
Comment (rauchg) [inline README.md]: `create_document` returns a `CreateDocumentRequest` which needs to be `execute`d to become a `Future`

**Comment by MindFlavor** [comment]
Great! Thank you!

**Pull request merged**